packages/viewer/src/modules/extensions/sections/SectionTool.ts

```diff
@@ -920,4 +959,37 @@ export class SectionTool extends Extension {
   ): box is OBB {
     return box instanceof OBB
   }
+
+  /**
+   * Snaps a quaternion to the nearest 15-degree grid.
+   * This is useful for rotation snapping.
+   * @param q The quaternion to snap.
+   * @returns The snapped quaternion.
+   */
+  protected snapQuaternionToGrid(q: Quaternion): Quaternion {
+    /** Convert quaternion to Euler angles using pooled object */
+    this.tempEuler.setFromQuaternion(q)
+
+    /** Snap each axis to 15-degree increments (π/12 radians) */
+    const snapAngle = Math.PI / 12 // 15 degrees
```

Comment:
I would turn this into an extension option and make it configurable so that we can have any snap value including none